From c1ff88c2c78e31d65abef335623fb343f73c322b Mon Sep 17 00:00:00 2001
From: LyraCalc OS <lyracalc-os@users.noreply.github.com>
Date: Thu, 22 Oct 2026 12:00:00 +0200
Subject: [PATCH 6/6] defconfig: USB WiFi, Bluetooth, LAN and serial adapters,
 zram, loopback sound

- WiFi: Realtek rtl8xxxu, Ralink rt2800usb, MediaTek mt7601u/mt76x0u/mt76x2u/
  mt7663u/mt7921u, Atheros ath9k_htc/carl9170, RTL8188EU
- Bluetooth: btusb Realtek and MediaTek, HIDP, UHID
- USB LAN and tethering: r8152, asix, ax88179, lan78xx, cdc_ether/eem/ncm,
  rndis_host, ipheth, smsc95xx, dm9601, sr9700
- USB serial: cdc-acm, ch341, cp210x, ftdi_sio
- zram, snd-aloop
---
 .../arch/arm/configs/rk3506_luckfox_defconfig | 54 ++++++++++++++++---
 1 file changed, 48 insertions(+), 6 deletions(-)

diff --git a/kernel-6.1/arch/arm/configs/rk3506_luckfox_defconfig b/kernel-6.1/arch/arm/configs/rk3506_luckfox_defconfig
index e3267585..af864cf9 100644
--- a/kernel-6.1/arch/arm/configs/rk3506_luckfox_defconfig
+++ b/kernel-6.1/arch/arm/configs/rk3506_luckfox_defconfig
@@ -81,7 +81,6 @@ CONFIG_BT_RFCOMM_TTY=y
 CONFIG_BT_BNEP=y
 CONFIG_BT_HCIBTUSB=m
 # CONFIG_BT_HCIBTUSB_BCM is not set
-# CONFIG_BT_HCIBTUSB_RTL is not set
 CONFIG_RFKILL=y
 CONFIG_RFKILL_RK=y
 CONFIG_DEVTMPFS=y
@@ -160,20 +159,15 @@ CONFIG_MOTORCOMM_PHY=m
 CONFIG_CANFD_ROCKCHIP=y
 CONFIG_CANFD_RK3576=y
 CONFIG_PPP=y
-# CONFIG_USB_NET_DRIVERS is not set
 # CONFIG_WLAN_VENDOR_ADMTEK is not set
-# CONFIG_WLAN_VENDOR_ATH is not set
 # CONFIG_WLAN_VENDOR_ATMEL is not set
 # CONFIG_WLAN_VENDOR_BROADCOM is not set
 # CONFIG_WLAN_VENDOR_CISCO is not set
 # CONFIG_WLAN_VENDOR_INTEL is not set
 # CONFIG_WLAN_VENDOR_INTERSIL is not set
 # CONFIG_WLAN_VENDOR_MARVELL is not set
-# CONFIG_WLAN_VENDOR_MEDIATEK is not set
 # CONFIG_WLAN_VENDOR_MICROCHIP is not set
 # CONFIG_WLAN_VENDOR_PURELIFI is not set
-# CONFIG_WLAN_VENDOR_RALINK is not set
-# CONFIG_WLAN_VENDOR_REALTEK is not set
 # CONFIG_WLAN_VENDOR_RSI is not set
 # CONFIG_WLAN_VENDOR_SILABS is not set
 # CONFIG_WLAN_VENDOR_ST is not set
@@ -392,4 +386,52 @@ CONFIG_BOOTPARAM_RCU_STALL_PANIC=y
 # CONFIG_FTRACE is not set
 CONFIG_DEBUG_USER=y
 # CONFIG_RUNTIME_TESTING_MENU is not set
+CONFIG_RTL8188EU=m
+CONFIG_ZRAM=m
 CONFIG_MMC_SPI=m
+CONFIG_USB_ACM=m
+CONFIG_USB_SERIAL=m
+CONFIG_USB_SERIAL_CH341=m
+CONFIG_USB_SERIAL_CP210X=m
+CONFIG_USB_SERIAL_FTDI_SIO=m
+CONFIG_SND_ALOOP=m
+CONFIG_SND_DRIVERS=y
+CONFIG_BT_HIDP=m
+CONFIG_UHID=m
+CONFIG_WLAN_VENDOR_REALTEK=y
+CONFIG_RTL8XXXU=m
+CONFIG_RTL8XXXU_UNTESTED=y
+CONFIG_WLAN_VENDOR_RALINK=y
+CONFIG_RT2X00=m
+CONFIG_RT2800USB=m
+CONFIG_RT2800USB_RT33XX=y
+CONFIG_RT2800USB_RT35XX=y
+CONFIG_RT2800USB_RT3573=y
+CONFIG_RT2800USB_RT53XX=y
+CONFIG_RT2800USB_RT55XX=y
+CONFIG_RT2800USB_UNKNOWN=y
+CONFIG_WLAN_VENDOR_MEDIATEK=y
+CONFIG_MT7601U=m
+CONFIG_MT76x0U=m
+CONFIG_MT76x2U=m
+CONFIG_MT7663U=m
+CONFIG_MT7921U=m
+CONFIG_WLAN_VENDOR_ATH=y
+CONFIG_ATH9K_HTC=m
+CONFIG_CARL9170=m
+CONFIG_BT_HCIBTUSB_MTK=y
+CONFIG_BT_HCIBTUSB_RTL=y
+CONFIG_USB_NET_DRIVERS=y
+CONFIG_USB_USBNET=m
+CONFIG_USB_RTL8152=m
+CONFIG_USB_NET_AX8817X=m
+CONFIG_USB_NET_AX88179_178A=m
+CONFIG_USB_LAN78XX=m
+CONFIG_USB_NET_CDCETHER=m
+CONFIG_USB_NET_CDC_EEM=m
+CONFIG_USB_NET_CDC_NCM=m
+CONFIG_USB_NET_RNDIS_HOST=m
+CONFIG_USB_IPHETH=m
+CONFIG_USB_NET_SMSC95XX=m
+CONFIG_USB_NET_DM9601=m
+CONFIG_USB_NET_SR9700=m
-- 
2.43.0

